flush-print/README.md

```diff
@@ -0,0 +1,64 @@
+# How to Flush the Output of the Python Print Function
+
+This directory contains code associated with the Real Python tutorial [How to Flush the Output of the Python Print Function](https://realpython.com/python-flush-print-output/).
+
+## Installation
+
+You don't need to install anything apart from a Python interpreter >=3.6 to run the example files.
+
+## Countdown Script Variations
+
+You can find all the code snippets that you've seen in the tutorial. For convenience, they're provided as separate files that all start with `countdown` and describe their differences with the rest of the file name. You can run each of the files separately without needing to edit the code, to see all the examples shown in the tutorial. Please visit the tutorial link for additional context.
+
+## Terminal Based Visual Progress Indicators
+
+Additionally, this directory also contains two scripts that show example implementations of terminal animations that you can build using `print()` and its parameters:
+
+- [`spinner.py`](spinner.py)
+- [`progress.py`](progress.py)
+
+Both of them utilize explicit flushing and you can learn more about how to build them in [Your Guide to the Python `print()` Function](https://realpython.com/python-print/).
+
+## Buffer Size Approximations
+
+Finally, if you're curious to dive somewhat deeper into the rabbit hole of buffering in Python, you can take a look at [`buffersize.py`](buffersize.py).
+
+There doesn't seem to be a way to get the size of the default buffer for stdout on your operating system directly through a Python object. If you know or find a way, please [contact the author](#about-the-author), I'd be very curious to know :)
+
+This script gives you a way to approximate the buffer size that Python uses when writing to stdout on your operating system and configuration. Note that this can be quite different between different setups.
+
+The approach to calculate it is somewhat manual:
+
+1. Run the script
+1. Note if the numbers that it prints pause in between before the script finishes execution
+  - If they pause, remember the last number that got printed before the pause
+  - If they don't pause, increase the value of `SLIGHTLY_TOO_LARGE_FOR_BUFFER` by `10_000` and start from the top
+
+Continue doing this until your script paused when printing the numbers, and you noted the number that displayed last during the pause. Once you have that number, subtract it from the value that `SLIGHTLY_TOO_LARGE_FOR_BUFFER` currently has to calculate your buffer size estimation, for example:
```

Comment:
Maybe we include the "manual" nature of this operation to be 100% crystal clear:

```suggestion
Continue doing this until your script paused when printing the numbers, and you noted the number that displayed last during the pause. Once you have that number, manually subtract it from the value that `SLIGHTLY_TOO_LARGE_FOR_BUFFER` currently has to calculate your buffer size estimation, for example:
```